### DISCUSSION TITLE
Enable iframe communication

### DISCUSSION THREAD

**Discussion by acontreras89** [grafana/grafana#54963] [General]
The ability to embed panels using iframes (although with [some limitations](https://github.com/grafana/grafana/discussions/54962)) is very powerfull. However, there is no way to respond to user interactions with the panel, since there is no communication at all between the host application and Grafana.

Such communication could be enabled using the [`postMessage` API](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage). If two-way communication would require a significant amount of effort, one-way communication might be quite straight-forward to implement. This would enable the host application to respond to user interactions with the panel.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
